src/libs/actions/IOU.js

```diff
@@ -561,7 +570,11 @@ function cancelMoneyRequest(chatReportID, iouReportID, type, moneyRequestAction)
     );
 
     const currentUserEmail = optimisticReportAction.actorEmail;
-    const updatedIOUReport = IOUUtils.updateIOUOwnerAndTotal(iouReport, currentUserEmail, amount, moneyRequestAction.originalMessage.currency, type);
+    const updatedIOUReport = IOUUtils.updateIOUOwnerAndTotal(iouReport, currentUserEmail, amount, moneyRequestAction.originalMessage.currency, CONST.IOU.REPORT_ACTION_TYPE.DELETE);
+
+    // Save original messages before updating the report in case the request fails and we need to rollback
+    const originalLastMessageText = chatReport.lastMessageText;
+    const originalLastMessageHtml = chatReport.lastMessageHtml;
```

Comment:
This is not related to the changes in this PR, but I strongly believe the code below should be corrected. Mutations are not good in a React project.

If you had not mutated `chatReport` below, these variables you are introducing here to keep the "original" values would not be needed.

Reply:
@luacmartins I think you don't need to clone the objects. We are using those values as read-only. I have suggested some similar change here https://github.com/Expensify/App/pull/17583#discussion_r1174327447 in code diff. 

_Please ignores the lines that remove `...optimisticReportAction` and add `pendingAction`._